unity/src/main/java/io/delta/unity/UCCatalogManagedClient.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.unity;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+
+import io.delta.kernel.ResolvedTable;
+import io.delta.kernel.TableManager;
+import io.delta.kernel.annotation.Experimental;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.annotation.VisibleForTesting;
+import io.delta.kernel.internal.files.ParsedLogData;
```

Comment:
Importing `internal` classes to me indicates we've not got the right API yet. Or do we expect that an OSS catalog would have to import and use these classes too? 

Basically it seems like we should strive to have a catalog client that only uses truly "public" kernel interfaces.

Reply:
> Importing internal classes to me indicates we've not got the right API yet.

Agreed! `Public ParsedLogData API` is on the backlog of refactors for this CCv2 work (https://github.com/delta-io/delta/issues/4573). Want me to implement than now / sooner rather than later?

Reply:
No, this is fine for now, but let's add a TODO in the code with a link to the issue